pkg/winutil/run_windows.go

```diff
@@ -0,0 +1,197 @@
+}
+
+// ShellExecuteAndWait is identical to https://pkg.go.dev/golang.org/x/sys/windows#ShellExecute,
+// except it customizes the SEE_MASK_NOCLOSEPROCESS fMask to allow the caller to wait for
+// execution to terminate.
+func ShellExecuteAndWait(hwnd HWND, verb, exe, args, wd LPCWSTR, nShowCmd int32) error {
+	i := &SHELLEXECUTEINFOW{
+		fMask:        SEE_MASK_NOCLOSEPROCESS,
+		hwnd:         hwnd,
+		lpVerb:       verb,
+		lpFile:       exe,
+		lpParameters: args,
+		lpDirectory:  wd,
+		nShow:        nShowCmd,
+	}
+	i.cbSize = DWORD(unsafe.Sizeof(*i))
+	return ShellExecuteEx(i)
+}
+
+// Some of the code below (particularly the error handling) was borrowed from
+// https://github.com/AllenDang/w32/blob/65507298e138d537445133ed145a1f2685782b34/shell32.go#L110
+// Copyright 2010-2012 The W32 Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the following comment:
+// Copyright (c) 2010-2012 The w32 Authors. All rights reserved.
+
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+// 1. Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+// 2. Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+// 3. The names of the authors may not be used to endorse or promote products
+//    derived from this software without specific prior written permission.
+
+// THIS SOFTWARE IS PROVIDED BY THE AUTHORS “AS IS” AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+// OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+// IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+// INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+// NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+// THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+func ShellExecuteEx(pExecInfo *SHELLEXECUTEINFOW) error {
+	ret, _, _ := procShellExecuteEx.Call(uintptr(unsafe.Pointer(pExecInfo)))
+	if ret == 1 && pExecInfo.fMask&SEE_MASK_NOCLOSEPROCESS != 0 {
+		s, e := syscall.WaitForSingleObject(syscall.Handle(pExecInfo.hProcess), syscall.INFINITE)
+		switch s {
+		case syscall.WAIT_OBJECT_0:
+			break
+		case syscall.WAIT_FAILED:
+			return os.NewSyscallError("WaitForSingleObject", e)
+		default:
+			return fmt.Errorf("Unexpected result from WaitForSingleObject")
```

Comment:
```suggestion
			return errors.New("Unexpected result from WaitForSingleObject")
```

Reply:
I just used `fmt.Errorf` to not need to import `errors`. The behavior should be the same but I can change it if you want